packages/core/src/amazonq/lsp/lspController.ts

```diff
@@ -0,0 +1,378 @@
+            return undefined
+        }
+        for (const version of manifest.versions) {
+            if (version.isDelisted) {
+                continue
+            }
+            if (!supportedLspServerVersions.includes(version.serverVersion)) {
+                continue
+            }
+            for (const t of version.targets) {
+                if (
+                    (t.platform === process.platform || (t.platform === 'windows' && process.platform === 'win32')) &&
+                    t.arch === process.arch
+                ) {
+                    for (const content of t.contents) {
+                        if (content.filename.startsWith('node') && content.hashes.length > 0) {
+                            content.serverVersion = version.serverVersion
+                            return content
+                        }
+                    }
+                }
+            }
+        }
+        return undefined
+    }
+
+    private async hashMatch(filePath: string, content: Content) {
+        const sha384 = await this.getFileSha384(filePath)
+        if ('sha384:' + sha384 !== content.hashes[0]) {
+            getLogger().error(
+                `LspController: Downloaded file sha ${sha384} does not match manifest ${content.hashes[0]}.`
+            )
+            fs.removeSync(filePath)
+            return false
+        }
+        return true
+    }
+
+    async downloadAndCheckHash(filePath: string, content: Content) {
+        await this._download(filePath, content.url)
+        const match = await this.hashMatch(filePath, content)
+        if (!match) {
+            return false
+        }
+        return true
+    }
+
+    async tryInstallLsp(context: vscode.ExtensionContext): Promise<boolean> {
+        try {
+            if (this.isLspInstalled(context)) {
+                getLogger().info(`LspController: LSP already installed`)
+                return true
+            }
+            const manifest: Manifest = (await this.downloadManifest()) as Manifest
+            const qserver = this.getQserverFromManifest(manifest)
+            const noderuntime = this.getNodeRuntimeFromManifest(manifest)
+            if (!qserver || !noderuntime) {
+                getLogger().info(`LspController: Did not find LSP URL for ${process.platform} ${process.arch}`)
+                return false
+            }
+
+            const tempFolder = await makeTemporaryToolkitFolder()
+            const zipFilePath = path.join(tempFolder, 'qserver.zip')
+            let downloadOk = await this.downloadAndCheckHash(zipFilePath, qserver)
+            if (!downloadOk) {
+                return false
+            }
+            const zip = new AdmZip(zipFilePath)
+            zip.extractAllTo(context.asAbsolutePath(path.join('resources')))
+            fs.removeSync(zipFilePath)
+
+            const noderuntimeFilePath = path.join(tempFolder, nodeBinName)
+            downloadOk = await this.downloadAndCheckHash(noderuntimeFilePath, noderuntime)
```

Comment:
prefer making this a different variable rather than sharing the same var with L233